tests/components/api/test_init.py

```diff
@@ -352,6 +352,30 @@ def listener(service_call):
     assert state["attributes"] == {"data": 1}
 
 
+async def test_api_call_service_timeout(
+    hass: HomeAssistant, mock_api_client: TestClient
+) -> None:
+    """Test if the API does not fail on long running services."""
+    test_value = []
+
+    fut = hass.loop.create_future()
+
+    async def listener(service_call):
+        """Wait and return after mock_api_client.post finishes."""
+        value = await fut
+        test_value.append(value)
+
+    hass.services.async_register("test_domain", "test_service", listener)
+
+    with patch("homeassistant.components.api.SERVICE_WAIT_TIMEOUT", 0):
+        await mock_api_client.post("/api/services/test_domain/test_service")
+        fut.set_result(1)
```

Comment:
Assert the `test_value` list here so we know that the service call hasn't completeted at this point, ie after the post call but before the future is done.

Reply:
But we already know it hasn't been completed, because we haven't provided the value for the future yet, and later we verify that the value is used. But if you believe an additional assert will be more readable, I will add it.